mllib/src/main/scala/org/apache/spark/ml/feature/package-info.java

```diff
@@ -0,0 +1,112 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * Feature transformers
+ *
+ * The `ml.feature` package provides common feature transformers that help convert raw data or
+ * features into more suitable forms for model fitting.
+ * Most feature transformers are implemented as {@link org.apache.spark.ml.Transformer}s, which
+ * transforms one {@link org.apache.spark.sql.DataFrame} into another, e.g.,
+ * {@link org.apache.spark.feature.HashingTF}.
+ * Some feature transformers are implemented as {@link org.apache.spark.ml.Estimator}}s, because the
+ * transformation requires some aggregated information of the dataset, e.g., document
+ * frequencies in {@link org.apache.spark.ml.feature.IDF}.
+ * For those feature transformers, calling {@link org.apache.spark.ml.Estimator#fit} is required to
+ * obtain the model first, e.g., {@link org.apache.spark.ml.feature.IDFModel}, in order to apply
+ * transformation.
+ * The transformation is usually done by appending new columns to the input
+ * {@link org.apache.spark.sql.DataFrame}, so all input columns are carried over.
+ *
+ * We try to make each transformer minimal, so it becomes flexible to assemble feature
+ * transformation pipelines.
+ * {@link org.apache.spark.ml.Pipeline} can be used to chain feature transformers, and
+ * {@link org.apache.spark.ml.feature.VectorAssembler} can be used to combine multiple feature
+ * transformations, for example:
+ *
+ * <pre>
+ * <code>
+ *   import java.util.Arrays;
+ *   import java.util.List;
+ *
+ *   import org.apache.spark.api.java.JavaRDD;
+ *   // Import factory methods provided by DataTypes.
+ *   import org.apache.spark.sql.types.DataTypes;
+ *   // Import StructType and StructField
+ *   import org.apache.spark.sql.types.StructType;
+ *   import org.apache.spark.sql.types.StructField;
+ *   import org.apache.spark.sql.DataFrame;
+ *   import org.apache.spark.sql.RowFactory;
+ *   import org.apache.spark.sql.Row;
+ *
+ *   import org.apache.spark.ml.feature.*;
+ *   import org.apache.spark.ml.Pipeline;
+ *   import org.apache.spark.ml.PipelineStage;
+ *   import org.apache.spark.ml.PipelineModel;
+ *
+ *  // a DataFrame with three columns: id (integer), text (string), and rating (double).
+ *  List<StructField> fields = Arrays.asList(
+ *      DataTypes.createStructField("id", DataTypes.IntegerType, false),
+ *      DataTypes.createStructField("text", DataTypes.StringType, false),
+ *      DataTypes.createStructField("rating", DataTypes.DoubleType, false));
+ *  StructType schema = DataTypes.createStructType(fields);
+ *  JavaRDD<Row> rowRDD = jsc.parallelize(
```

Comment:
Not in this PR, we could add `createDataFrame(List, StructType)` to `SQLContext`: https://issues.apache.org/jira/browse/SPARK-10630